comp/networkpath/npscheduler/npschedulerimpl/pathteststore/pathteststore.go

```diff
@@ -0,0 +1,144 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package pathteststore handle pathtest storage
+package pathteststore
+
+import (
+	"sync"
+	time "time"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/networkpath/npscheduler/npschedulerimpl/common"
+)
+
+var timeNow = time.Now
+
+// PathtestContext contains Pathtest information and additional flush related data
+type PathtestContext struct {
+	Pathtest *common.Pathtest
+
+	nextRunTime       time.Time
+	runUntilTime      time.Time
+	lastFlushTime     time.Time
+	lastFlushInterval time.Duration
+}
+
+// LastFlushInterval returns last flush interval
+func (p *PathtestContext) LastFlushInterval() time.Duration {
+	return p.lastFlushInterval
+}
+
+// SetLastFlushInterval sets last flush interval
+func (p *PathtestContext) SetLastFlushInterval(lastFlushInterval time.Duration) {
+	p.lastFlushInterval = lastFlushInterval
+}
+
+// PathtestStore is used to accumulate aggregated pathtestContexts
+type PathtestStore struct {
+	logger log.Component
```

Comment:
It's not a hard requirement, but since we are moving toward components in general, it seems that we should the logger component when possible instead of using the global logger.

Reply:
There is just a lot of repetition with `pathtest`. The packet is named `pathteststore`, so this reads as `pathteststore.PathtestStore`, instead of something like `pathteststore.Store` (even that "stutters"). This doesn't of course impact functionality, but aids in code readability a lot.

Reply:
Good points. I did some simplification here e763b7627e

Let me know if you see more things.

Note: I didn't renamed `PathtestContext` to `Context`, since `Context` alone can be ambiguous and not easily recognisable.